Pinta.Core/Classes/AsyncEffectRenderer.cs

```diff
@@ -264,16 +262,19 @@ void RenderTile (int renderId, int threadId, int tileIndex)
 		// Update bounds to be shown on next expose.
 		lock (updated_lock) {
 			if (is_updated) {
-				updated_x1 = Math.Min (bounds.X, updated_x1);
-				updated_y1 = Math.Min (bounds.Y, updated_y1);
-				updated_x2 = Math.Max (bounds.X + bounds.Width, updated_x2);
-				updated_y2 = Math.Max (bounds.Y + bounds.Height, updated_y2);
+				updated_area = RectangleI.FromLTRB (
+					left: Math.Min (bounds.X, updated_area.X),
+					top: Math.Min (bounds.Y, updated_area.Y),
+					right: Math.Max (bounds.X + bounds.Width - 1, updated_area.Right),
+					bottom: Math.Max (bounds.Y + bounds.Height - 1, updated_area.Bottom)
+				);
 			} else {
 				is_updated = true;
-				updated_x1 = bounds.X;
-				updated_y1 = bounds.Y;
-				updated_x2 = bounds.X + bounds.Width;
-				updated_y2 = bounds.Y + bounds.Height;
+				updated_area = new (
```

Comment:
Can this just be `updated_area = bounds` ?

Reply:
Thanks for noticing this, I've now shortened the code according to your suggestion